### ISSUE TITLE
Error: incompatible character encodings: UTF-8 and ASCII-8BIT

### ISSUE THREAD

**Issue by ablyler** [Homebrew/homebrew#49757]
Trying to pull in some pull request in homebrew-php, and I'm getting the following error:

```
Andys-MacBook-Pro :: ~/Code/homebrew-php ‹master› » brew pull https://github.com/Homebrew/homebrew-php/pull/2921
==> Fetching patch
######################################################################## 100.0%
Error: incompatible character encodings: UTF-8 and ASCII-8BIT
Please report this bug:
    https://git.io/brew-troubleshooting
/usr/local/Library/Homebrew/vendor/okjson.rb:361:in `[]='
/usr/local/Library/Homebrew/vendor/okjson.rb:361:in `unquote'
/usr/local/Library/Homebrew/vendor/okjson.rb:285:in `strtok'
/usr/local/Library/Homebrew/vendor/okjson.rb:251:in `tok'
/usr/local/Library/Homebrew/vendor/okjson.rb:214:in `lex'
/usr/local/Library/Homebrew/vendor/okjson.rb:45:in `decode'
/usr/local/Library/Homebrew/utils/json.rb:10:in `load'
/usr/local/Library/Homebrew/cmd/pull.rb:325:in `current_versions_from_info_external'
/usr/local/Library/Homebrew/cmd/pull.rb:91:in `block in pull'
/usr/local/Library/Homebrew/cmd/pull.rb:40:in `each'
/usr/local/Library/Homebrew/cmd/pull.rb:40:in `pull'
/usr/local/Library/brew.rb:83:in `<main>'
```

**Comment by apjanke** [comment]
Adjusted formatting: the triple-backticks for code block quotes need to be on their own lines.

**Comment by apjanke** [comment]
I can reproduce in my OS X 10.9 setup.

```
$ brew pull https://github.com/Homebrew/homebrew-php/pull/2921
Warning: Current branch is test-pull-for-49757: do you need to pull inside master?
==> Fetching patch
######################################################################## 100.0%
Error: incompatible character encodings: UTF-8 and ASCII-8BIT
Please report this bug:
    https://git.io/brew-troubleshooting
/usr/local/Library/Homebrew/vendor/okjson.rb:361:in `[]='
/usr/local/Library/Homebrew/vendor/okjson.rb:361:in `unquote'
/usr/local/Library/Homebrew/vendor/okjson.rb:285:in `strtok'
/usr/local/Library/Homebrew/vendor/okjson.rb:251:in `tok'
/usr/local/Library/Homebrew/vendor/okjson.rb:214:in `lex'
/usr/local/Library/Homebrew/vendor/okjson.rb:45:in `decode'
/usr/local/Library/Homebrew/utils/json.rb:10:in `load'
/usr/local/Library/Homebrew/cmd/pull.rb:324:in `current_versions_from_info_external'
/usr/local/Library/Homebrew/cmd/pull.rb:90:in `block in pull'
/usr/local/Library/Homebrew/cmd/pull.rb:39:in `each'
/usr/local/Library/Homebrew/cmd/pull.rb:39:in `pull'
/usr/local/Library/brew.rb:83:in `<main>'
```

This shouldn't be happening, I think. That `current_versions_from_info_external` is a new function introduced in #48472; maybe I missed an encoding edge case there. I'll try to have a look at this.

**Comment by ablyler** [comment]
@apjanke i think you are right, I created a fix in PR #49775 that fixes it.

**Comment by apjanke** [comment]
I think I see why it's happening for this PR but not others. The `php70` formula's caveats contains the string `\n\n✩✩✩✩ PHP CLI ✩✩✩✩\n`. Most other formulae stick to ASCII-compatible characters.
